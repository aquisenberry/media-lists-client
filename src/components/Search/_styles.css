.search {
  width: 71%;
  margin: 2rem auto;

  display: flex;
  align-items: center;
  justify-content: center;

  padding: 0 1.75rem;
  border-radius: 3rem;
  background: var(--color-primary-1);
  border: solid 1px var(--color-primary-2);
}

.search__input {
  flex: 1;
  border: none;
  font-size: 1.5rem;
  font-family: var(--font-raleway);
  font-weight: 500;
  padding-right: 1rem;

  outline: none;
  color: var(--color-utility);
  background: var(--color-primary-1);
}

.search__image {
  width: 35px;
  height: 35px;
  cursor: pointer;
} 

 .search__button {
  padding: 20px 40px;
  border-radius: 0.5rem;
  margin-left: 15px;
  color: var(--color-secondary);
  font-family: var(--font-secondary);
  font-weight: 900;
  letter-spacing: 0.75px;
  font-size: 1.25rem;
  cursor: pointer;
  background: var(--color-primary-1);
}