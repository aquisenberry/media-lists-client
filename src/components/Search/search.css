/* .search {
  width: 71%;
  margin: 4rem 0 2rem;

  display: flex;
  align-items: center;
  justify-content: center;

  padding: 1.5rem 1.75rem;
  border-radius: 3rem;
  background: #1f2123;
  -webkit-box-shadow: 5px 5px 7px #1c1d1f, -5px -5px 7px #222527;
  box-shadow: 5px 5px 7px #1c1d1f, -5px -5px 7px #222527;
}

.search input {
  flex: 1;
  border: none;
  font-size: 1.5rem;
  font-family: var(--font-raleway);
  font-weight: 500;
  padding-right: 1rem;

  outline: none;
  color: #a1a1a1;
  background: #1f2123;
}

.search img {
  width: 35px;
  height: 35px;
  cursor: pointer;
} */

/* .search button {
  padding: 20px 40px;
  border-radius: 0.5rem;
  margin-left: 15px;
  color: #a1a1a1;
  font-family: var(--font-raleway);
  font-weight: 900;
  letter-spacing: 0.75px;
  font-size: 1.25rem;
  cursor: pointer;

  background: #1f2123;
  -webkit-box-shadow: 5px 5px 7px #1c1d1f, -5px -5px 7px #222527;
  box-shadow: 5px 5px 7px #1c1d1f, -5px -5px 7px #222527;
} */

/* 
@media screen and (max-width: 600px) {
    .search {
      padding: 1rem 1.75rem;
      width: 100%;
    }
  
    .search input {
      font-size: 1rem;
    }
  
    .search img {
      width: 20px;
      height: 20px;
    }
  } */
