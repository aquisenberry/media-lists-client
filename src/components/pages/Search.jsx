import React from 'react';
import MediaList from '../MediaList';
import {useSelector, useDispatch} from 'react-redux'
import { setSearchTerm, searchAllMedia } from '../../redux/actions/external';
import SearchIcon from '../../assets/images/searchIcon.svg'


function Search() {
    const dispatch = useDispatch()
    const searchTerm = useSelector((state) => state.external.searchTerm)
    const movies = useSelector((state) => state.external.movies)
    const shows = useSelector((state) => state.external.shows)
    const videoGames = useSelector((state) => state.external.videoGames)
    const boardGames = useSelector((state) => state.external.boardGames)
    const books = useSelector((state) => state.external.books)
    return (
        <>
            <h1>Find Media</h1>
            <div className="search">
                <input className="search__input"placeholder="Search for Movies"
                    value={searchTerm}
                    onChange={(e) => dispatch(setSearchTerm(e.target.value))}
                />
                <button>
                    <img 
                        src={SearchIcon}
                        alt= "Search"
                        onClick={() => dispatch(searchAllMedia(searchTerm))}
                    />
                </button>
            </div>

            <MediaList title="Movies" mediaList={movies} />
            <MediaList title="TV Shows" mediaList={shows} />
            <MediaList title="Video Games" mediaList={videoGames} />
            <MediaList title="Board Games"mediaList={boardGames} />
            <MediaList title="Books"mediaList={books} />
            
        </>
    );
}

export default Search;