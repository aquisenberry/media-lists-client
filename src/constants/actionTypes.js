// export const CREATE = 'CREATE'
// export const UPDATE = 'UPDATE'
// export const DELETE = 'DELETE'
// export const FETCH_ALL = 'FETCH_ALL'
// export const LIKE = 'LIKE'

// External ActionTypes
export const FETCHED_MOVIES = 'FETCHED_MOVIES'
export const FETCHED_SHOWS = 'FETCHED_SHOWS'
export const FETCHED_VIDEO_GAMES = 'FETCHED_VIDEO_GAMES'
export const FETCHED_BOARD_GAMES = 'FETCHED_BOARD_GAMES'
export const FETCHED_BOOKS = 'FETCHED_BOOKS'
export const FETCHED_MEDIA_DETAILS = 'FETCHED_MEDIA_DETAILS'
export const UPDATED_SEARCH_TERM = 'UPDATED_SEARCH_TERM'
export const FETCHED_ALL_MEDIA = 'FETCHED_ALL_MEDIA'

// page ActionTypes
export const ADDED_MEDIA = 'ADDED_MEDIA'
export const SELECTED_MEDIA = 'SELECTED_MEDIA'
export const UPDATED_CONTEXT_MENU_ANCHOR = 'UPDATED_CONTEXT_MENU_ANCHOR'
export const UPDATED_CONTEXT_MENU_VISIBILITY = 'UPDATED_CONTEXT_MENU_VISIBILITY'
export const SET_HOVERED_ELEMENT = 'SET_HOVERED_ELEMENT'