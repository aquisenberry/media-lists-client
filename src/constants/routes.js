export const LISTS = 'lists'
export const MOVIES = 'movies'
export const SHOWS = 'shows'
export const IGDB = 'igdb'
export const ATLAS = 'atlas'
export const BOOKS = 'books'