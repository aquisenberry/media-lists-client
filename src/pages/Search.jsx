import React from 'react'
import MediaList from '../components/MediaList'
import SearchBar from '../components/Search'
import {useSelector} from 'react-redux'


function Search() {
    const movies = useSelector((state) => state.external.movies)
    const shows = useSelector((state) => state.external.shows)
    const videoGames = useSelector((state) => state.external.videoGames)
    const boardGames = useSelector((state) => state.external.boardGames)
    const books = useSelector((state) => state.external.books)

    
    return (
        <>
            <h1>Find Media</h1>
            <SearchBar />
            <MediaList title="Movies" mediaList={movies} />
            <MediaList title="TV Shows" mediaList={shows} />
            <MediaList title="Video Games" mediaList={videoGames} />
            <MediaList title="Board Games"mediaList={boardGames} />
            <MediaList title="Books"mediaList={books} />
            
        </>
    );
}

export default Search;