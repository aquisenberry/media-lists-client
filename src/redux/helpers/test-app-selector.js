
const state = {
    external: {
        searchTerm: "",
        movies:[],
        shows:[],
        videoGames: [],
        boardGames: [],
        books: [],
        mediaDetails: {}
    },
    page: {
        media: [],
        contextMenuAnchor: {
            x:0,
            y:0
        },
        showContextMenu: false,
        contextTarget:{},
        contextType: false
    }
}

export const testUseAppSelector = (f) => f(state)