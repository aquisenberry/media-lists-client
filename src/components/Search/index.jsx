
import React, {useEffect} from 'react';
import { setSearchTerm, searchAllMedia } from '../../redux/actions/external';
import { useAppDispatch, useAppSelector } from '../../redux/helpers/redux-hooks';
import SearchIcon from '../../assets/images/searchIcon.svg'

import './_styles.css'
const SearchBar = () => {

    const dispatch = useAppDispatch()
    const searchTerm = useAppSelector((state) => state.external.searchTerm)

    const handleKeyDown = (e) =>{
        if (e.key === "Enter"){
            dispatch(searchAllMedia(searchTerm))
        }
    }

    useEffect(() => {
        dispatch(searchAllMedia(searchTerm))
    }, [dispatch])

    return (
        <div className="search">
                <input className="search__input"placeholder="Search for Media"
                    value={searchTerm}
                    onChange={(e) => dispatch(setSearchTerm(e.target.value))}
                    onKeyDown={handleKeyDown}
                />
                <button className="search__button">
                    <img 
                        className="search__image"
                        src={SearchIcon}
                        alt= "Search"
                        onClick={() => dispatch(searchAllMedia(searchTerm))}
                    />
                </button>
            </div>
    )
}


export default SearchBar