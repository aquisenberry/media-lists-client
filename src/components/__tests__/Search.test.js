import renderer from 'react-test-renderer'
import {getByRole, render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Search from '../Search'
import {useAppDispatch, useAppSelector} from '../../redux/helpers/redux-hooks'
import { testUseAppSelector } from '../../redux/helpers/test-app-selector'
import { UPDATED_SEARCH_TERM } from '../../constants/actionTypes'


jest.mock('../../redux/helpers/redux-hooks')

describe("Search", () =>{
    
    const dispatch = jest.fn()
    beforeEach(() =>{
        useAppDispatch.mockImplementation(() => dispatch);
        useAppSelector.mockImplementation(testUseAppSelector)
    })
    afterEach(() =>{
        jest.clearAllMocks()
    })
    it('has no markup regressions', () =>{
        const component = renderer.create(
            <Search/>
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('calls setSearchTerm when the input is typed in', () =>{
        render(<Search/>)
        const input = screen.getByRole('textbox')
        userEvent.type(input, 'Hello')
        expect(dispatch.mock.calls.length).toBe(5)
        expect(dispatch).toHaveBeenNthCalledWith(1,{type: UPDATED_SEARCH_TERM, payload: 'H'});
    })
    it('calls searchAllMedia when the input is selected and [Enter] is pressed', () =>{
        render(<Search/>)
        const input = screen.getByRole('textbox')
        userEvent.type(input, '{enter}')
        expect(dispatch.mock.calls.length).toBe(1)
        // there is a more meaningful test to write here 
        // that verifies the correct action creator was dispatched
        // but I can't figure it out at the moment
    })
})

