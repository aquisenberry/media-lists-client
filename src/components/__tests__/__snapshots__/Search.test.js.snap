// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has no markup regressions 1`] = `
<div
  className="search"
>
  <input
    className="search__input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search for Media"
    value=""
  />
  <button
    className="search__button"
  >
    <img
      alt="Search"
      className="search__image"
      onClick={[Function]}
      src="searchIcon.svg"
    />
  </button>
</div>
`;
